projects/angular-calendar/src/modules/month/calendar-month-row.component.ts

```diff
@@ -149,54 +137,52 @@ export class CalendarMonthRowComponent implements OnChanges {
-      )
-      .findIndex(
-        (date) => new Date(date.toDateString()).valueOf() === note.end.valueOf()
-      );
-    // start et end dans le row en cours
-    if (indexEndDate !== -1 && indexStartDate !== -1) {
-      width = (indexEndDate - indexStartDate + 1) * (100 / 7);
+    const oneDayInMs = 1000 * 60 * 60 * 24;
+    const weekEndDate = new Date(
+      this.daysSliced[6].date.getTime() + oneDayInMs - 1
+    );
+    const noteLengthInDays = this.dateDiffIndays(note.start, note.end);
+    const fromWeekStartToNoteStartInDays = this.dateDiffIndays(
+      this.daysSliced[0].date,
+      note.start
+    );
+    const fromWeekStartToNoteEndInDays = this.dateDiffIndays(
+      this.daysSliced[0].date,
+      note.end
+    );
+    const fromNoteStartToWeekEndInDays = this.dateDiffIndays(
+      note.start,
+      weekEndDate
+    );
+
+    const noteStartOnWeek =
+      fromWeekStartToNoteStartInDays > 0 && fromWeekStartToNoteStartInDays <= 7;
+    const noteEndOnWeek =
+      fromWeekStartToNoteEndInDays > 0 && fromWeekStartToNoteEndInDays <= 7;
+
+    if (noteStartOnWeek && noteEndOnWeek) {
+      width = noteLengthInDays * (100 / 7);
     }
-    // start dans le row en cours et pas le end
-    else if (indexStartDate !== -1 && indexEndDate === -1) {
-      const daysNumberFromNoteStart = this.dateDiffIndays(
-        note.start,
-        this.timezoneManager.reverseTz(this.daysSliced[6].date, this.timezone)
-      );
-      width = (daysNumberFromNoteStart + 1) * (100 / 7);
+    if (!noteStartOnWeek && noteEndOnWeek) {
+      width = fromWeekStartToNoteEndInDays * (100 / 7);
     }
-    // end dans le row en cours et pas le start
-    else if (indexStartDate === -1 && indexEndDate !== -1) {
-      const daysNumberToNoteEnd = this.dateDiffIndays(
-        this.timezoneManager.reverseTz(this.daysSliced[0].date, this.timezone),
-        note.end
-      );
-      width = (daysNumberToNoteEnd + 1) * (100 / 7);
+    if (noteStartOnWeek && !noteEndOnWeek) {
+      width = fromNoteStartToWeekEndInDays * (100 / 7);
     }
-    // start et end sont en dehors du row en cours
-    else if (indexStartDate === -1 && indexEndDate === -1) {
+    if (!noteStartOnWeek && !noteEndOnWeek) {
       width = 100;
     }
     return 'calc(' + width + '% - 5px)';
   }
 
   dateDiffIndays(date1: Date, date2: Date): number {
-    return Math.floor(
-      (Date.UTC(date2.getFullYear(), date2.getMonth(), date2.getDate()) -
-        Date.UTC(date1.getFullYear(), date1.getMonth(), date1.getDate())) /
-        (1000 * 60 * 60 * 24)
-    );
+    const oneDayInMs = 1000 * 60 * 60 * 24;
+    const differenceInMs = date2.getTime() - date1.getTime();
+    const differenceInMsAbs = Math.abs(differenceInMs);
+    const differenceInDays =
+      Math.floor(differenceInMsAbs / oneDayInMs) +
+      (differenceInMsAbs % oneDayInMs > 0 ? 1 : 0);
+    return differenceInMs > 0 ? differenceInDays : differenceInDays * -1;
```

Comment:
There is a flaw in the method since we consider day will have a 24 hours duration whereas some might end 23 hours and others will end 25 hours thanks to DST.
Since the diff we calculate is day and DST is 1 hour we could get to valid result replacing `floor` by `round`.

But IMHO we shouldn't have need to this method at all.

for the ref the test that is failing with current implementation : 
```
            // With timezone set to Europe/London
            const startDate = new Date('2023-10-26 23:00:00 +00:00'); // 2023-10-27 00:00:00 London
            const endDate = new Date('2023-10-31 00:00:00 +00:00'); // 2023-10-31 00:00:00 London
            const dayDiff = dateDiffIndays(startDate, endDate);
            expect(dayDiff).toEqual(4); // Fail returns 5
```

Reply:
look at the : `noteLengthInDaysOnWeek` function 
(as it was before) i base calculation on findIndex (i just made more accurate condition for finding indexes than it was before)